src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -32,6 +32,23 @@ internal static class IntrinsicFunctions
         private static readonly Lazy<Regex> RegistrySdkRegex = new Lazy<Regex>(() => new Regex(@"^HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Microsoft SDKs\\Windows\\v(\d+\.\d+)$", RegexOptions.IgnoreCase));
 #endif // FEATURE_WIN32_REGISTRY
 
+        /// <summary>
+        /// Resolve the location of the NuGet.Frameworks assembly
+        /// </summary>
+        private static Assembly GetNuGetAssembly()
+        {
+            var assemblyDirectory = IsRunningFromVisualStudio() ?
+                Path.Combine(GetVsInstallRoot(), "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet") :
+                BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory;
+            return Assembly.LoadFile(Path.Combine(assemblyDirectory, "NuGet.Frameworks.dll"));
```

Comment:
What's the behavior if this fails? Can we present a nice error like "hey, couldn't find NuGet"?